src/resolver/resolve_path.zig

```diff
@@ -623,6 +623,15 @@ pub fn normalizeStringGeneric(
 ) []u8 {
     return normalizeStringGenericT(u8, path_, buf, allow_above_root, separator, isSeparator, preserve_trailing_slash);
 }
+
+fn separatorAdapter(comptime T: type, func: anytype) fn (T) bool {
```

Comment:
this is fine but later this would be legendary to make into a generic wrapper for

"wrap any function with type"

Reply:
yeah I agree, could probably live in the bun namespace. Only problem is that zig doesn't support variadics, so it can't be fully generalized 🥲

Reply:
see std.meta.ArgsTuple

it’s like variadics but worse